Source/player.h

```diff
@@ -313,6 +313,38 @@ struct PlayerStruct {
 	Uint8 *_pDData;
 	Uint8 *_pBData;
 
+	/**
+	 * @brief Gets the most valuable item out of all the player's items that match the given predicate.
+	 * @param itemPredicate The predicate used to match the items.
+	 * @return The most valuable item out of all the player's items that match the given predicate, or 'nullptr' in case no
+	 * matching items were found.
+	 */
+	template <typename TPredicate>
+	const ItemStruct *GetMostValuableItem(TPredicate itemPredicate) const
+	{
+		auto getMostValuableItem = [](const ItemStruct *begin,
+		                               const ItemStruct *end,
+		                               TPredicate itemPredicate,
+		                               const ItemStruct *&mostValuableItem) {

```

Comment:
To make this more readable:
Replace the `const ItemStruct *&mostValuableItem` argument with `const ItemStruct *mostValuableItem` and return the item instead.

Reply:
I initially made this function return the item, however then it became convoluted to "find the most valuable item among belt, inventory and equipment".

By passing a writable pointer, I'm able to "accumulate" the greatest one from all 3 by just calling them in sequence.

I guess you want me to pass the reference to the "maximum thus far" but still return it? I'll take a look.